### PR TITLE
feat: support custom class component

### PR DESCRIPTION
**Pull request by agileago**

### CODE REVIEW

File: src/mount.ts [hunk 1]
```diff
@@ -73,6 +74,25 @@ function getInstanceOptions(
   return resultOptions
 }
 
+// Standard Class component
```
Comment (cexbrayat) [inline src/mount.ts:77]: nit: I would remove "Standard" (here and everywhere), as I suspect most people won't write this component like this. I would use "Class component (without vue-class-component)" instead.

File: src/mount.ts [hunk 2]
```diff
@@ -73,6 +74,25 @@ function getInstanceOptions(
...
+export function mount<V>(
+  originalComponent: {
+    new (...args: any[]): V
+    __vccOpts: any
+  },
+  options?: MountingOptions<any> & Record<string, any>
+): VueWrapper<ComponentPublicInstance<V>>
+
+// Standard Class component - props
```
Comment (cexbrayat) [inline src/mount.ts:86]: Same here "Class component (without vue-class-component) - props"

File: test-dts/mount.d-test.ts [hunk 3]
```diff
@@ -211,6 +218,78 @@ class ClassComponent extends Vue {
 expectError(mount(ClassComponent, {}).vm.changeMessage())
 mount(ClassComponent, {}).vm.changeMessage('')
 
+// region standard class component implement
+class StandardClassComponent<Props extends {} = {}> {
```
Comment (cexbrayat) [inline test-dts/mount.d-test.ts:222]: `class CustomClassComponent`

Comment (cexbrayat) [reply test-dts/mount.d-test.ts:222]: (remove standard in the comment as well)

File: test-dts/mount.d-test.ts [hunk 4]
```diff
@@ -211,6 +218,78 @@ class ClassComponent extends Vue {
...
+  render() {
+    return h('div', `hello world ${this.count.value}`)
+  }
+}
+
+// @ts-expect-error changeCount expects an argument
+expectError(mount(NoPropClassComponent, {}).vm.changeCount())
+mount(NoPropClassComponent, {}).vm.changeCount(2)
+
+interface WithProp_Props {
```
Comment (cexbrayat) [inline test-dts/mount.d-test.ts:265]: We usually don't use underscores in interface names. Maybe just `ComponentProps`?

**Review by cexbrayat — CHANGES_REQUESTED** [review]
Just a few name to change and we'll be good to merge 👍

**Comment by agileago** [comment]
@cexbrayat thanks for your patient

**Review by lmiller1990 — APPROVED** [review]

**Pull request merged**